proposals/build-provenance-and-code-signing-for-homebrew.md

```diff
@@ -110,3 +110,55 @@ For formulae with verifiable source artifacts, the signed bottle attestation des
 ### macOS executable code signing
 
 Because this proposal only concerns bottles and not their individual executable members, it should have no effect on Homebrew's current use or any future uses of "ad-hoc" executable signatures (which are produced not for authenticity, but because macOS on Apple Silicon requires all executables to be signed).
+
+## Threat model
+
+### Scope
+
+The scope of this proposal is *authenticated build provenance* for Homebrew bottles.
+
+As an end state: when a user performs `brew install foo`, the bottle corresponding to `foo` will be installed *if and only if* it has a valid signature defined over an *attestation* for that bottle. That signature in turn is considered valid *if and only if* it is both cryptographically valid  and* corresponds to the tap repository identity that `foo` was built in (e.g. `homebrew/homebrew-core`).
+
+### Attacker models
+
+All attacker models listed below are remote models. Local attackers are not modeled, under Homebrew's pre-existing operating assumption that a local attacker capable of modifying files or running code has already won. This assumption is comparable to the client integrity assumption in the Web PKI (i.e., that the client's root of trust is not itself compromised).
+
+These models are not intended to be exhaustive.
+
+#### Compromise of bottle storage
+
+**Scenario**: Mallory is able to circumvent Homebrew's ordinary bottle building and publishing workflows and is able to upload malicious bottles to a tap's bottle storage.
+
+**Mitigation**: With *just* upload access to the bottle storage, Mallory is unable to forge valid signatures for her malicious bottles. `brew install $compromised-bottle` fails to install due to a missing or mismatched signature, and users who attempt to install `$compromised-bottle` are loudly notified.
+
+**Outcome**: Mallory is **unable** to force an install of a compromised bottle. Mallory remains undetected.
```

Comment:
What do you mean by undetected? I read this as an undetected attack would be in the case of a successful compromise.

Reply:
Ah, as I keep reading, I see why this is mentioned. It might be worth mentioning transparency as an overall goal early in the threat model.

Reply:
Yep, I meant in a transparency sense. Thanks for pointing that out, I can emphasize that!

Reply:
Added another summary point for this, emphasizing that *just* being undetected isn't a "win" state for Mallory (she must also successfully perform her attack). Let me know if you think I should elaborate further!